paimon-format/src/main/java/org/apache/paimon/format/orc/writer/OrcBulkWriter.java

```diff
@@ -67,4 +89,43 @@ public void finish() throws IOException {
         flush();
         writer.close();
     }
+
+    @Override
+    public long length() throws IOException {
+        long estimateMemory = treeWriter.estimateMemory();
+
+        long dataLength = 0;
+        List<StripeInformation> stripes =
+                Collections.unmodifiableList(OrcUtils.convertProtoStripesToStripes(this.stripes));
+        if (!stripes.isEmpty()) {
+            StripeInformation stripeInformation = stripes.get(stripes.size() - 1);
+            dataLength =
+                    stripeInformation != null
+                            ? stripeInformation.getOffset() + stripeInformation.getLength()
+                            : 0;
+        }
+
+        // This value is estimated, not actual.
+        return (long)
+                Math.ceil(
+                        dataLength
+                                + (estimateMemory + (long) rowBatch.size * avgRowByteSize) * 0.2);
```

Comment:
By default, rowBatch is 1000, and the rolling is required every 1000 records, my question is: can we check rolling during flushBatch?

So the `avgRowByteSize` can be removed.

Reply:
It is OK to let it be 1024.

Reply:
@JingsongLi  I have removed `avgRowByteSize`  and make the check interval equal with the size of `rowBatch`. Please help to review again.